1차 스프린트에서 설계한 테이블을 모두 엔티티와 매핑한다

alex-gibeom opened the pull request:
## 개요

- close: #4 

## 작업 요약

- 1차 스프린트에서 설계한 테이블을 모두 엔티티 객체와 매핑한다

## 작업 내용

> (체크 박스를 모두 선택해야 merge가 활성화 됩니다.)

- [x]  회원 테이블
- [x]  회원 찜 정보 테이블
- [x] 매장 정보 테이블
- [x] 매장 사진 정보 테이블
- [x] 예약 정보 테이블
- [x] 리뷰 정보 테이블
- [x] CS 요청 내역 테이블

<br>

## 참고 자료
- https://www.erdcloud.com/d/uykGfNjFfDtZDvzpk

<img src="https://user-images.githubusercontent.com/59248326/212026299-8558e6c1-6d5e-4ec8-aab0-0cf15a381143.png" width="100%" />

alex-gibeom committed:
feat(#4): 1차 스프린트 테이블을 모두 엔티티와 매핑한다

giibeom commented:
## 코드리뷰 내용 정리

- [ ] 등록자와 수정자도 필요하다
   - BaseTimeEntity -> BaseEntity 
- [ ] 전화번호나 예약시간 같은 자릿수가 정해져있는(변하지 않는) 형식의 컬럼은 DB 타입 사이즈를 딱 맞게 조절하는 것이 좋다
   - 매장 영업시간 : VARCHAR(50) -> VARCHAR(11)
- [ ] 컬럼에 테이블 prefix 붙이는 것과 기타 제약 조건을 붙이는 것을 다시 생각해보자
   - 제약 조건은 주석 느낌이여서 DBA가 바꿔도 제대로 반영을 못할 상황이 생길 수 있다 (주석으로 인한 혼란 발생)   
   - 붙일거면 이유를 설명할 수 있어야 한다
- [ ] NOT NULL 제약 조건은 꼭 필요한 컬럼에만 설정하자
   - 막 붙이면 테스트 코드 짤때도, 이벤트 리스너 관련 작업에서도 여러모로 피곤하다
- [ ] 대표 이미지 컬럼은 NMJ_STORE 테이블 컬럼으로 이관한다
   - 대표 이미지가 아닌 데이터가 훨씬 많기에 비효율적
- [ ] 멤버의 default 값은 직접 명시해주는 것이 직관적이다 (builder안에 직접 기본 값 선언)
- [ ] 디미터의 법칙을 준수하며, 묻지말고 시키자(Tell, Don't Ask)

giibeom commented:
> * [ ]  등록자와 수정자도 필요하다
>   * BaseTimeEntity -> BaseEntity

- Spring Security 적용 예정은 당장 없으므로 등록자와 수정자 컬럼은 보류
- https://www.baeldung.com/database-auditing-jpa#4-auditing-the-author-of-changes-with-spring-security

alex-gibeom committed:
feat(#6): 엔티티 매핑 관련 코드 리뷰 내용을 적용한다

- https://github.com/giibeom/project-NMJ-renewal/pull/6#issuecomment-1380588037
